tests/integration/workflows/test_submission_workflow.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Tests for submission workflows."""
+
+from six import StringIO
+import mock
+
+from inspirehep.modules.workflows.tasks.submission import (
+    submit_rt_ticket
+)
+
+
+class MockLog(object):
+    def __init__(self):
+        self._log = StringIO()
+
+    def info(self, message):
+        self._log.write(message)
+
+
+class MockObj(object):
+
+    log = MockLog()
+
+    def __init__(self, id_, data, extra_data):
+        self.id = id_
+        self.data = data
+        self.extra_data = extra_data
+
+
+def mock_get_rt_instance():
+
+    class MockRtInstance(object):
+        def create_ticket(self, *args, **kwargs):
+            return 1
+
+    return MockRtInstance()
+
+
+@mock.patch('inspirehep.modules.workflows.tasks.submission.get_instance',
+            side_effect=mock_get_rt_instance)
+def test_submit_rt_ticket(mocker_rt_instance, small_app):
+    """test that a ticket is sent to the RT service"""
+
+    body = ''
+    queue = ''
+    context = {}
+    obj = MockObj(1, {}, {})
+    ticket_id_key = 'ticket_id'
+    small_app.config['PRODUCTION_MODE'] = True
+
+    assert submit_rt_ticket(obj,
+                            queue,
+                            context.get('subject'),
+                            body,
+                            'mail@ma.il',
+                            ticket_id_key)
+
+    assert mocker_rt_instance.call_count == 1
```

Comment:
Take a look at this: https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html